[vulkan-headers, vulkan-hpp ,vulkan-memory-allocator] Update to the latestion version

chrismile opened the pull request:
**Describe the pull request**

- #### What does your PR fix?  
  This PR updates vulkan-headers, vulkan-hpp and vulkan-memory-allocator to more recent versions (1.2.184 for vulkan-headers and vulkan-hpp).

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  All, no.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes, except "Make separate Pull Requests per port". I decided to include all three updates in one PR, as they all concern Vulkan ports and an update with respect to Vulkan 1.2.184.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes.

chrismile committed:
Updated vulkan-headers, vulkan-hpp and vulkan-memory-allocator to new versions.

chrismile committed:
Updated the port ashes to fix compilation errors with the new Vulkan headers. Added vk_video/ to the list of copied header files in the port vulkan-headers.

JonLiu1993 commented on ports/ashes/vcpkg.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ashes",
-  "version-string": "2021-02-09",
+  "version-string": "2021-06-18",
```
```suggestion
  "version-date": "2021-06-18",
```

JonLiu1993 commented on ports/vulkan-headers/vcpkg.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "vulkan-headers",
-  "version-string": "1.2.157",
+  "version-string": "1.2.184",
```
```suggestion
  "version-semver": "1.2.184",
```

chrismile committed:
Apply suggestions from code review regarding version strings

Update version strings in ports/ashes/vcpkg.json and ports/vulkan-headers/vcpkg.json.

Co-authored-by: JonLiu1993 <63675417+JonLiu1993@users.noreply.github.com>

chrismile committed:
Updated versioning information of modified ports.

JonLiu1993 commented:
@chrismile ,Please run the command "./vcpkg x-add-version --all --overwrite-version" and commit the change again.

chrismile committed:
Update version strings after changes in vcpkg.json files.

chrismile commented:
Thanks a lot for your feedback! All requested changes should now be integrated in the PR.
